src/serve/hosting.ts

```diff
@@ -1,22 +1,19 @@
-"use strict";
+import clc = require("cli-color");
+import { FirebaseError } from "../error";
+import * as utils from "../utils";
+import * as detectProjectRoot from "../detectProjectRoot";
+import functionsProxy from "../hosting/functionsProxy";
+import cloudRunProxy from "../hosting/cloudRunProxy";
+const superstatic = require("superstatic").server;
+import { implicitInit, TemplateServerResponce } from "../hosting/implicitInit";
+import { initMiddleware } from "../hosting/initMiddleware";
+import { normalizedHostingConfigs } from "../hosting/normalizedHostingConfigs";
 
-var clc = require("cli-color");
-var { FirebaseError } = require("../error");
+const MAX_PORT_ATTEMPTS = 10;
+let _attempts = 0;
+let server: any;
 
-var superstatic = require("superstatic").server;
-var utils = require("../utils");
-var detectProjectRoot = require("../detectProjectRoot");
-var implicitInit = require("../hosting/implicitInit");
-var initMiddleware = require("../hosting/initMiddleware");
-var functionsProxy = require("../hosting/functionsProxy").default;
-var cloudRunProxy = require("../hosting/cloudRunProxy").default;
-var normalizedHostingConfigs = require("../hosting/normalizedHostingConfigs");
-
-var MAX_PORT_ATTEMPTS = 10;
-var _attempts = 0;
-var server;
-
-function _startServer(options, config, port, init) {
+function _startServer(options: any, config: any, port: number, init: TemplateServerResponce): void {
```

Comment:
Should `config` here use the `HostingConfig` type from the other file?  If so should we move that somewhere more central?

Reply:
> Should config here use the HostingConfig type from the other file?
Yeah. I'll change that code.

> If so should we move that somewhere more central?
maybe yes.
this `config` dependents supertest. so, this is  more better  if` @types/supertest` is exists 🤔